internal/archiver/archiver.go

```diff
@@ -264,6 +264,9 @@ func (arch *Archiver) trackItem(item string, previous, current *restic.Node, s I
 // nodeFromFileInfo returns the restic node from an os.FileInfo.
 func (arch *Archiver) nodeFromFileInfo(snPath, filename string, meta ToNoder, ignoreXattrListError bool) (*restic.Node, error) {
 	node, err := meta.ToNode(ignoreXattrListError)
+	if node == nil {
+		return nil, fmt.Errorf("ToNode returned nil node for %v: %w", filename, err)
+	}
```

Comment:
This should only happen when ToNode returns an error, so

```suggestion
	if err != nil {
		return nil, err
	}
```

Reply:
only either `node==nil` or `node==nil&&err!=nil` work here. If prefer the latter, as `node==nil` + `err==nil` is a clear bug that must be fixed.